de.fu_berlin.inf.dpp/test/framework/stf/src/de/fu_berlin/inf/dpp/stf/server/rmi/superbot/component/view/impl/Views.java

```diff
@@ -24,12 +30,34 @@ public static Views getInstance() {
     return INSTANCE;
   }
 
+  private boolean forceUseHtmlGui = false;
+
+  /* Use this method to force the STF to use the HTML-GUI.
+   * This can be removed when the old GUI is fully replaced.
+   * */
+  @Override
+  public void forceUseHtmlGui() throws RemoteException {
+    if (Saros.useHtmlGui() == false) {
+      throw new RuntimeException(
+          "HTML-GUI is not enabled. Please set saros.swtbrowser propertie to true.");
+    }
+    forceUseHtmlGui = true;
+  }
+
   @Override
   public ISarosView sarosView() throws RemoteException {
-    SWTWorkbenchBot bot = new SWTWorkbenchBot();
-    RemoteWorkbenchBot.getInstance().openViewById(VIEW_SAROS_ID);
-    bot.viewByTitle(VIEW_SAROS).show();
-    return SarosView.getInstance().setView(bot.viewByTitle(VIEW_SAROS));
+    if (Saros.useHtmlGui() && this.forceUseHtmlGui) {
```

Comment:
It's confusing to have a setter method and the underlying field with the same name. Maybe something like `private boolean shouldTestHtmlGui` and `public void runTestsOnHtmlGui()`?

By the way: Who is supposed to call `forceUseHtmlGui()`? If it's going to be the same party that calls `sarosView()`, why introduce a field and a new method instead of simply a boolean parameter for `public ISarosView sarosView(boolean testHtmlGui)`?

Reply:
I also thought about it with Kelvin. The disadvantage is that you have to adjust the `sarosView()` Method in every test. Or do you mean to introduce a second `sarosView(boolean testHtmlGui)` method?

Reply:
I don't have a preference. But since it's Java Code and we control all the code, even the first option would boil down to a single refactoring.

Reply:
I renamed the variable and the method to your suggestion.